src/lib/operators/aggregate_hash.cpp

```diff
@@ -239,7 +242,30 @@ void AggregateHash::_aggregate() {
       }
     }
 
-    // Now that we have the data structures in place, we can start the actual work
+    // Now that we have the data structures in place, we can start the actual work. We want to fill
+    // keys_per_chunk[chunk_id][chunk_offset] with something that uniquely identifies the group into which that
+    // position belongs. There are a couple of options here (cf. AggregateHash::_on_execute):
+    //
+    // 0 GROUP BY columns:   No partitioning needed; we don't reach this point because of the check for
+    //                       EmptyAggregateKey above
+    // 1 GROUP BY column:    The AggregateKey is one dimensional, i.e., the same as AggregateKeyEntry
+    // > 1 GROUP BY columns: The AggregateKey is multi-dimensional. The value in
+    //                       keys_per_chunk[chunk_id][chunk_offset] is subscripted with the index of the GROUP BY
+    //                       columns (not the same as the GROUP BY column_id)
+    //
+    // To generate a unique identifier, we create a map from the value found in the respective GROUP BY column to
+    // a unique uint64_t. The value 0 is reserved for NULL.
+    //
+    // This has the cost of a hashmap lookup and potential insert for each row and each GROUP BY column. There are
+    // some cases in which we can avoid this. These make use of the fact that we can only have 2^64 - 2*2^32 values
+    // in a table (due to INVALID_VALUE_ID and INVALID_CHUNK_OFFSET limiting the range of RowIDs).
+    //
+    // (1) For types smaller than AggregateKeyEntry, such as int32_t, their value range can be immediately mapped into
+    //     uint64_t. We cannot do the same for int64_t because we need to account for NULL values.
+    // (2) For strings not longer than five characters, there are 1+2^(1*8)+2^(2*8)+2^(3*8)+2^(4*8) potential values.
+    //     We can immediately map these into a numerical representation by reinterpreting their byte storage as an
+    //     integer. The calculation is described below.
```

Comment:
I was first not sure how you obtain the information that no string is longer than 5 chars. Maybe clear that up  here? Since (1) will be used for the whole vector, I'd thought (2) does the same.

Reply:
"Note that this is done on a per-string basis and does not require all strings in the given column to be that short."

Better?

Reply:
Jep, would help a lot.